src/core/agent/services/connectionService.ts

```diff
@@ -395,7 +390,20 @@ class ConnectionService extends AgentService {
     } else if (!operation.done) {
       throw new Error(ConnectionService.FAILED_TO_RESOLVE_OOBI);
     }
-    const oobi = { ...operation, alias };
+
+    const connectionId =
+      operation.done && operation.response
+        ? operation.response.i
+        : new URL(url).pathname.split("/oobi/").pop()!.split("/")[0];
+
+    const keriaConnection = await this.props.signifyClient
+      .contacts()
+      .update(connectionId, {
+        alias,
+        groupCreationId: groupId,
+        createdAt: new Date().toISOString(),
+      });
+    const oobi = { ...operation, ...keriaConnection };
```

Comment:
I don't see where `keriaConnection` is being used. Can't we just `return operation;`?

Reply:
Because we need to return the KERIA connection to create the connection metadata in the `connectByOobiUrl` function so I return that

Reply:
Right, the response type is confusing because in `connectByOobiUrl` it's called `operation`. Can we just return: `{ op: operation, connection: keriaConnection }`? The variable shouldn't be called `oobi` here either.